ludwig/datasets/titanic/__init__.py

```diff
@@ -0,0 +1,68 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2019 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import os
+import pandas as pd
+from ludwig.datasets.base_dataset import BaseDataset, DEFAULT_CACHE_LOCATION
+from ludwig.datasets.mixins.kaggle import KaggleDownloadMixin
+from ludwig.datasets.mixins.load import CSVLoadMixin
+
+
+def load(cache_dir=DEFAULT_CACHE_LOCATION, split=False):
+    dataset = Titanic(cache_dir=cache_dir)
+    return dataset.load(split=split)
+
+
+class Titanic(CSVLoadMixin, KaggleDownloadMixin, BaseDataset):
+    """The Titanic dataset.
+
+    This pulls in an array of mixins for different types of functionality
+    which belongs in the workflow for ingesting and transforming training data into a destination
+    dataframe that can fit into Ludwig's training API.
+    """
+    config: dict
```

Comment:
We don't need to define all these variables here since we can infer their existence from the parent classes.

Reply:
Removed